File: 30 Days of Code/Day 18 Queues and Stacks/Solution.py
class Solution:
    def __init__( self ):
        self.__stack = [ ]
        self.__queue = [ ]

    def pushCharacter( self, c ):
        self.__stack.append( c )

    def enqueueCharacter( self, c ):
        self.__queue.append( c )

    def popCharacter( self ):
        return self.__stack.pop( )

    def dequeueCharacter( self ):
        return self.__queue.pop( 0 )

s = input( )
obj = Solution( )
for idx in len( s ):
    obj.pushCharacter( s[ i ] )
    obj.enqueueCharacter( s[ i ] )
isPalindrome = True
for _ in range( len( s ) / 2):
    if obj.popCharacter( ) != obj.dequeueCharacter( ):
        isPalindrome = False
        break

if isPalindrome:
    print( "The word, " + s + ", is a palindrome." )
else:
    print( "The word, " + s + ", is not a palindrome." )

File: 30 Days of Code/Day 18 Queues and Stacks/Solution.cs
using System;
using System.Collections.Generic;

class Solution {

    Stack< char > stack = new Stack< char >( );
    Queue< char > queue = new Queue< char >( );

    public void pushCharacter( char c ) {
        stack.Push( c );
    }

    public void enqueueCharacter( char c ) {
        queue.Enqueue( c );
    }

    public char popCharacter( ) {
        return stack.Pop( );
    }

    public char dequeueCharacter( ) {
        return queue.Dequeue( );
    }

    public static void Main( String[ ] args ) {
        String s = Console.ReadLine( );
        Solution obj = new Solution( );

        foreach( char c in s ) {
            obj.pushCharacter( c );
            obj.enqueueCharacter( c );
        }

        bool isPalindrome = true;
        
        for( int idx = 0; idx < s.Length / 2; idx++ ) {
            if( obj.popCharacter( ) != obj.dequeueCharacter( ) ) {
                isPalindrome = false;
                break;
            }
        }

        if( isPalindrome ) {
            Console.WriteLine( "The word, {0}, is a palindrome.", s );
        } else {
            Console.WriteLine( "The word, {0}, is not a plaindrome.", s );
        }
    }
}